README.md

```diff
@@ -28,15 +28,70 @@ In your browser, head to `http://localhost:9393/hello/danny`. Now go to `http://
 
 ### How Dynamic Routes Work:
 
-+ Dynamic routes will create a key-value pair in the `params` hash. 
-+ The key is determined by the symbol (for example `:name`) that you add to your route. 
-+ The associated value will be the content in the url provided by the user.
-+ Use the data provided in the same way you would a form - by accessing the key value pair in the `params` hash.
-+ You can receive multiple pieces of data through a dynamic route by separating the content with a forward slash. For example, `get '/addnumbers/:number1/:number2` would give you a `params` hash with two key-value pairs (number1 and number2).
+It's important to note that in Sinatra, a route is simply a HTTP method/verb that is paired with a URL-matching pattern. When your Sinatra application receives a request, it will match that route to a specific controller action that matches that URL pattern.
+
+The best way to explain routes is by going through an example. Our application is a medicine application that has an array containing three instances of a Medicine class.
+
+Here's our array:
+
+```ruby
+all_the_medicines = [ 
+  #<Medicine:0x007fb739b1af88 @id=1, @name="penicillin" @group="antibiotic">, 
+  #<Medicine:0x007fb739b1af88 @id=2, @name="advil" @group="anti-inflammatory">, 
+  #<Medicine:0x007fb739b1af88 @id=3, @name="benadryl" @group="anti-histamine">
+]
+```
+
+Our application gets a request :`GET /medicines/1`. What happens here?
+
+The first thing Sinatra does is try to match the request to a specific controller action. The controller action it would match is as follows: `get '/medicines/:id'`. Once the request has been matched to the controller action, it then executes the code inside of the controller action block, as shown below:
+
+```ruby
+# medicines_controller.rb
+get '/medicines/:id' do
+  @medicine = all_the_medicines.select do |medicine|
+    medicine.id == params[:id]
+  end.first
+  erb :'/medicines/show.html'
+end
+```
+
+Let's run through this specific scenario. The HTTP request verb, `GET` matches the `get` method in our controller. The `/medicines` path in the HTTP request matches the `/medicines` path in our controller method. Finally, the `17`, which is an `id` parameter that's being passed into the path, matches the controller's expectation for an `id` parameter to be passed in place of `:id`. 
+
+### URL Params
+
+A URl parameter is a variable whose values are set dynamically in a page's URL, and can be accessed by its template (your view) and its data sources (your models). It's a very similar concept to a dynamic url.
```

Comment:
Only accessible from your controller really. Your controller can then choose to pass it to the view or model